tests/__init__.py

```diff
@@ -1,10 +1,14 @@
 import logging
+import os
+
 from pprint import pformat
 from deepdiff import DeepDiff
 
 logging.basicConfig()
 logging.root.setLevel(logging.INFO)
 
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
```

Comment:
I defined this so that we can use this absolute path when loading test files from the `test/logs` directory. The previous method worked fine when running `make test` from the root project directory, but running individual tests via PyCharm would mean that the test, by default, starts with the directory of that test file as the current working directory. This is easy to change in configuration, but it would be nicer to just have it "solved" so that the correctness of our unit tests do not depend on being run from a particular directory.

I went ahead and updated all `Path`s for those test files accordingly as well!

Reply:
This works when executing the test directly in VSCode. Does it not work in PyCharm?
```
% git diff
diff --git a/tests/test_parse_local.py b/tests/test_parse_local.py
index d55359b..34feee4 100644
--- a/tests/test_parse_local.py
+++ b/tests/test_parse_local.py
@@ -16,7 +16,7 @@ def get_parsed_log(event_log_path):
 
 
 def test_simple_databricks_log():
-    path = Path(ROOT_DIR, "logs", "databricks.json").resolve()
+    path = Path("tests", "logs", "databricks.json").resolve()
     parsed_app = get_parsed_log(path)
     assert all(key in parsed_app for key in PARSED_KEYS), PARSED_KEYS_MISSING_MESSAGE
     assert (

```

Reply:
Seems that by default, PyCharm puts you in the same directory that the test is in. So for me locally, this means `//Users/gseyffert/develop/spark_log_parser/tests` instead of `/Users/gseyffert/develop/spark_log_parser/`

Reply:
The path as-written works just fine for me when I run all of the tests from the root project directory. The issue only pops up when trying to run an individual test like -

<img width="395" alt="Screenshot 2022-12-22 at 7 08 08 PM" src="https://user-images.githubusercontent.com/117235708/209262971-f24138e9-64b9-433d-80a0-b0965ab1f1cf.png">

Reply:
Wow, seems like a big issue for PyCharm users. Works as expected in VSCode:
<img width="610" alt="Screen Shot 2022-12-23 at 11 15 49 AM" src="https://user-images.githubusercontent.com/5589236/209396511-fd32efa0-f0cd-41af-84af-426f785a4f24.png">

Reply:
I wonder if there is a way to configure PyCharm to change this 🤔

Reply:
I'd expect any IDE to support running tests directly while keeping the "root" directory as the working directory...

Reply:
And you would be correct! I was able to find this setting in PyCharm, but I've never had to use it before. So indeed, this does seem like something that we could remove. Though perhaps we'd want to leave a note in the README for PyCharm users about project config,